## 1. Print Numbers from 1 to 20
### Question
### Write a program to print numbers from 1 to 20 using a while loop.
### Explanation
#### • Start from 1
#### • Print the number
#### • Increase the value by 1 each time
#### • Stop when number becomes greater than 20

In [1]:
num = 1

while num <= 20:
    print(num)
    num += 1

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20


## 2. Print Even Numbers
### Question
### Write a program to print all even numbers between 1 and 50 using a whileloop.
### Explanation
#### • Even numbers are divisible by 2
#### • Use loop from 1 to 50
#### • Check condition using if num % 2 == 0

In [2]:
num = 1

while num <= 50:
    if num % 2 == 0:
        print(num)
    num += 1

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50


## 3. Sum of Natural Numbers
### Question
### Write a program to calculate the sum of first n natural numbers.
### Explanation

### Example:
#### 1 + 2 + 3 + 4 + 5 = 15
#### • Take value of n
#### • Use loop from 1 to n
#### • Add each number into a sum variable

In [3]:
n = int(input("Enter n value: "))

sum = 0
num = 1

while num <= n:
    sum += num
    num += 1

print("Sum =", sum)

Enter n value:  15


Sum = 120


## 4. Reverse a Number
### Question
### Write a program to reverse a number using a while loop.
### Explanation
#### Example:
#### 1234 → 4321
#### Steps:
#### • Take last digit using % 10
#### • Add digit into reverse number
#### • Remove last digit using // 10

In [4]:
num = int(input("Enter number: "))

reverse = 0

while num > 0:
    digit = num % 10
    reverse = reverse * 10 + digit
    num = num // 10

print("Reverse =", reverse)

Enter number:  25


Reverse = 52


## 5. Count Digits in a Number
### Question
### Write a program to count total digits in a number.
### Explanation
#### Example:
#### 56789 → 5 digits
#### Steps:
#### • Use loop until number becomes 0
#### • Remove one digit each time
#### • Increase counter by 1

In [5]:
num = int(input("Enter number: "))

count = 0

while num > 0:
    num = num // 10
    count += 1

print("Total digits:", count)

Enter number:  45


Total digits: 2


## 6. Palindrome Number
### Question
### Write a program to check whether a number is palindrome or not.
### Explanation
#### Palindrome means same from both sides.
#### Example:
#### 121 → Palindrome
#### 1331 → Palindrome
#### Steps:
#### • Reverse the number
#### • Compare with original number

In [7]:
num = int(input("Enter number: "))

original = num
reverse = 0

while num > 0:
    digit = num % 10
    reverse = reverse * 10 + digit
    num = num // 10

if original == reverse:
    print("Palindrome Number")
else:
    print("Not Palindrome")

Enter number:  66


Palindrome Number


## 7. Armstrong Number
### Question
### Write a program to check whether a number is Armstrong number.
### Explanation
#### Example:
#### 153
#### Calculation:
#### 13 + 53 + 33 = 153
#### If sum equals original number → Armstrong number

In [10]:
num = int(input("Enter number: "))

original = num
sum = 0

while num > 0:
    digit = num % 10
    sum += digit ** 3
    num = num // 10

if sum == original:
    print("Armstrong Number")
else:
    print("Not Armstrong")

Enter number:  153


Armstrong Number


## 8. Password Checker
### Question
### Write a program to continuously ask password until correct password is entered.
### Explanation
#### • Store one correct password
#### • Use while loop
#### • Keep asking user input
#### • Stop when password matches

In [11]:
correct_password = "python123"

while True:
    password = input("Enter password: ")

    if password == correct_password:
        print("Password Correct")
        break
    else:
        print("Wrong Password")

Enter password:  python123


Password Correct


## 9. Guessing Game
### Question
### Write a program for number guessing game.
### Explanation
#### • Store one secret number
#### • Ask user to guess
#### • If wrong → ask again
#### • If correct → print success message

In [12]:
secret = 7

while True:
    guess = int(input("Guess number: "))

    if guess == secret:
        print("Correct Guess!")
        break
    else:
        print("Try Again")

Guess number:  7


Correct Guess!


## 10. Multiplication Table
### Question
### Write a program to print multiplication table of a number.
### Explanation
#### Example for 5:
#### 5 x 1 = 5
#### 5 x 2 = 10
#### ...
#### • Start from 1
#### • Multiply each value with given number
#### • Repeat till 10

In [13]:
num = int(input("Enter number: "))

i = 1

while i <= 10:
    print(num, "x", i, "=", num*i)
    i += 1

Enter number:  15


15 x 1 = 15
15 x 2 = 30
15 x 3 = 45
15 x 4 = 60
15 x 5 = 75
15 x 6 = 90
15 x 7 = 105
15 x 8 = 120
15 x 9 = 135
15 x 10 = 150


## 11. Fibonacci Series
### Question
### Write a program to print Fibonacci series.
#### Explanation
#### Series:
#### 0 1 1 2 3 5 8 13 ...
#### Each next number is sum of previous two numbers.

In [14]:
n = int(input("Enter number of terms: "))

a = 0
b = 1
count = 0

while count < n:
    print(a)
    
    c = a + b
    a = b
    b = c
    
    count += 1

Enter number of terms:  15


0
1
1
2
3
5
8
13
21
34
55
89
144
233
377


## 12. Factorial of Number
### Question
### Write a program to find factorial of a number.
#### Explanation
#### Example:
#### 5! = 5 × 4 × 3 × 2 × 1 = 120
#### • Multiply numbers one by one using loop

In [15]:
num = int(input("Enter number: "))

fact = 1

while num > 0:
    fact *= num
    num -= 1

print("Factorial =", fact)

Enter number:  15


Factorial = 1307674368000


## 13. Menu Driven Calculator
### Question
### Create calculator using while loop and if elif else.
### Explanation
#### Menu Example:
##### 1. Add
##### 2. Subtract
##### 3. Multiply
##### 4. Divide
##### 5. Exit
#### • User selects option
#### • Perform operation
#### • Continue until exit option chosen

In [16]:
while True:

    print("1.Add")
    print("2.Subtract")
    print("3.Multiply")
    print("4.Divide")
    print("5.Exit")

    choice = int(input("Enter choice: "))

    if choice == 5:
        break

    a = int(input("Enter first number: "))
    b = int(input("Enter second number: "))

    if choice == 1:
        print(a+b)

    elif choice == 2:
        print(a-b)

    elif choice == 3:
        print(a*b)

    elif choice == 4:
        print(a/b)

    else:
        print("Invalid Choice")

1.Add
2.Subtract
3.Multiply
4.Divide
5.Exit


Enter choice:  1
Enter first number:  1020
Enter second number:  20


1040
1.Add
2.Subtract
3.Multiply
4.Divide
5.Exit


Enter choice:  5


## 14. ATM System
### Question
### Create simple ATM program.
### Explanation
### Options:
#### • Check balance
#### • Deposit money
#### • Withdraw money"
#### • Exit
##### Use loop so menu repeats again and again.

In [17]:
balance = 5000

while True:

    print("\n1.Check Balance")
    print("2.Deposit")
    print("3.Withdraw")
    print("4.Exit")

    choice = int(input("Enter choice: "))

    if choice == 1:
        print("Balance:", balance)

    elif choice == 2:
        amount = int(input("Deposit amount: "))
        balance += amount
        print("Deposited")

    elif choice == 3:
        amount = int(input("Withdraw amount: "))

        if amount <= balance:
            balance -= amount
            print("Withdraw Successful")
        else:
            print("Insufficient Balance")

    elif choice == 4:
        break


1.Check Balance
2.Deposit
3.Withdraw
4.Exit


Enter choice:  1


Balance: 5000

1.Check Balance
2.Deposit
3.Withdraw
4.Exit


Enter choice:  1


Balance: 5000

1.Check Balance
2.Deposit
3.Withdraw
4.Exit


Enter choice:  4


## 15. Sum Until User Enters 0
### Question
### Take numbers from user continuously and stop when user enters 0.
### Explanation
#### • Add all numbers into sum
#### • When input becomes 0 → stop loop
#### • Print final sum

In [19]:
sum = 0

while True:

    num = int(input("Enter number: "))

    if num == 0:
        break

    sum += num

print("Total Sum:", sum)

Enter number:  56
Enter number:  23
Enter number:  10
Enter number:  5
Enter number:  4
Enter number:  2
Enter number:  0


Total Sum: 100


## 16. Leap Year Checker Repeatedly
### Question
### Check leap year repeatedly until user types "stop".
### Explanation
#### • Ask year from user
#### • Check leap year condition
#### • Continue until "stop" entered
### Leap year condition:
#### • Divisible by 4
#### • Not divisible by 100 OR divisible by 400

In [20]:
while True:

    year = input("Enter year or stop: ")

    if year == "stop":
        break

    year = int(year)

    if year % 4 == 0 and (year % 100 != 0 or year % 400 == 0):
        print("Leap Year")
    else:
        print("Not Leap Year")

Enter year or stop:  2025


Not Leap Year


Enter year or stop:  2024


Leap Year


Enter year or stop:  stop


## 17. Star Pattern
### Question
### Print star pattern using while loop.
### Explanation
### Pattern:
#### *
#### **
#### ***
#### ****
#### *****
### • Start from 1 star
### • Increase stars each iteration

In [21]:
row = 1

while row <= 5:
    print("*" * row)
    row += 1

*
**
***
****
*****


## 18. Prime Numbers Between 1 to 100
### Question
### Print all prime numbers from 1 to 100.
### Explanation
#### Prime number:
#### • Divisible only by 1 and itself
### Example:
#### 2, 3, 5, 7, 11 ...
### Use nested loop to check divisibility.

In [22]:
num = 2

while num <= 100:

    count = 0
    i = 1

    while i <= num:
        if num % i == 0:
            count += 1
        i += 1

    if count == 2:
        print(num)

    num += 1

2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


## 19. Student Marks Average
### Question
### Keep taking student marks until negative number entered.
### Explanation
#### • Add all marks
#### • Count total students
#### • Stop on negative input
#### • Find average:
### average = total / count

In [24]:
total = 0
count = 0

while True:

    marks = int(input("Enter marks: "))

    if marks < 0:
        break

    total += marks
    count += 1


average = total / count

print("Average Marks:", average)

Enter marks:  5
Enter marks:  24
Enter marks:  65
Enter marks:  23
Enter marks:  326
Enter marks:  66
Enter marks:  36
Enter marks:  65
Enter marks:  56
Enter marks:  -25


Average Marks: 74.0


## 20. Login System with 3 Attempts
### Question
### Create login system with maximum 3 attempts.
### Explanation
#### • Store correct username and password
#### • Ask user input
#### • If correct → Login successful
#### • If wrong → decrease attempts
#### • After 3 wrong attempts → block account

In [25]:
correct_username = "admin"
correct_password = "1234"

attempts = 3

while attempts > 0:

    username = input("Enter username: ")
    password = input("Enter password: ")

    if username == correct_username and password == correct_password:
        print("Login Successful")
        break

    else:
        attempts -= 1
        print("Wrong Login")
        print("Attempts left:", attempts)


if attempts == 0:
    print("Account Blocked")

Enter username:  admin
Enter password:  1234


Login Successful
